File: Ejercicio 3/ejercicio3.py
class Pelicula:
    def __init__(self, titulo):
        self.titulo = titulo

class Usuario:
    def __init__(self, username):
        self.username = username
        self.favoritos = []

    def agregar_a_favoritos(self, pelicula):
        self.favoritos.append(pelicula)


if __name__ == "__main__":
   
    p1 = Pelicula("Spider-Man")
    p2 = Pelicula("Batman")
    p3 = Pelicula("Avengers")

  
    u1 = Usuario("Andres_CC")
    u2 = Usuario("Maria_IA")

   
    u1.agregar_a_favoritos(p1)
    u1.agregar_a_favoritos(p2)
    u2.agregar_a_favoritos(p1)

    for user in [u1, u2]:
        print(f"Favoritos de {user.username}:")
        for fav in user.favoritos:
            print(f"  > {fav.titulo}")
        print("-" * 25)

File: Ejercicio 3/ejercicio3.java
import java.util.ArrayList;
import java.util.List;

class Pelicula {
    private String titulo;
    public Pelicula(String titulo) { this.titulo = titulo; }
    public String getTitulo() { return titulo; }
}

class Usuario {
    private String username;
    private List<Pelicula> favoritos;

    public Usuario(String username) {
        this.username = username;
        this.favoritos = new ArrayList<>();
    }

    public void agregarAFavoritos(Pelicula p) {
        this.favoritos.add(p);
    }

    public String getUsername() { return username; }
    public List<Pelicula> getFavoritos() { return favoritos; }
}

public class Main {
    public static void main(String[] args) {
      
        Pelicula p1 = new Pelicula("Inception");
        Pelicula p2 = new Pelicula("Interstellar");
        Pelicula p3 = new Pelicula("The Dark Knight");

        
        Usuario user1 = new Usuario("Cinefilo_99");
        Usuario user2 = new Usuario("TechUser");

    
        user1.agregarAFavoritos(p1);
        user1.agregarAFavoritos(p3);
        user2.agregarAFavoritos(p1); // Ambos pueden tener la misma favorita

       
        imprimirFavoritos(user1);
        imprimirFavoritos(user2);
    }

    public static void imprimirFavoritos(Usuario u) {
        System.out.println("Lista de " + u.getUsername() + ":");
        for (Pelicula p : u.getFavoritos()) {
            System.out.println(" - Favoriteada: " + p.getTitulo());
        }
        System.out.println("---------------------------");
    }
}
